citylearn/reward_function.py

```diff
@@ -307,4 +307,134 @@ def calculate(self, observations: List[Mapping[str, Union[int, float]]]) -> List
         reward = reward*np.reshape(self.coefficients, (len(self.coefficients), 1))
         reward = reward.sum(axis=0).tolist()
 
-        return reward
\ No newline at end of file
+        return reward
+
+
+class V2GPenaltyReward(MARL):
+    """Rewards with considerations for electric vehicle charging behaviours in a V2G setting.
+    Note that this function rewards/penalizes only the electric vehicle part. For a comprehensive reward strategy
+    please use one of the super classes or rewrite your own
+
+    Parameters
+    ----------
+    env_metadata: Mapping[str, Any]:
+        General static information about the environment.
+    """
+
+    def __init__(self, env_metadata: Mapping[str, Any],
+                 peak_percentage_threshold=0.10,
+                 ramping_percentage_threshold=0.10,
+                 peak_penalty_weight=20,
+                 ramping_penalty_weight=15,
+                 energy_transfer_bonus=10,
+                 window_size=6,
+                 penalty_no_car_charging=-5,
+                 penalty_battery_limits=-2,
+                 penalty_soc_under_5_10=-5,
+                 reward_close_soc=10,
+                 reward_self_ev_consumption=5,
+                 community_weight=0.2,
+                 reward_extra_self_production=5):
+        super().__init__(env_metadata)
+
+        # Setting the parameters
+        self.PEAK_PERCENTAGE_THRESHOLD = peak_percentage_threshold

```

Comment:
I don't think these parameters should be defined as constants though; they are set using instance 'variables' so i will rename then using lowercase to reflect that.